File: src/js/algorithms/bubble-sort/bubble-sort.js
/* 
  Bubble Sort
  Average Time complexity: O(n^2) where n is the number of values

*/

const { swap } = require("../helpers/helpers");

// sorts lowest to highest
function bubbleSort(array) {
  for (let i = 0; i < array.length; i++) {
    for (let j = i + 1; j < array.length; j++) {
      if (array[i] > array[j]) {
        swap(i, j, array);
      }
    }
  }
  return array;
}

module.exports = { bubbleSort };

File: src/go/src/algorithms/bubble-sort/bubble-sort.go
package bubblesort

func bubbleSort(slice []int) []int {
	swapped := true
	for swapped {
		swapped = false
		for i := 0; i < len(slice)-1; i++ {
			if slice[i] > slice[i+1] {
				swap(slice, i, i+1)
				swapped = true
			}
		}
	}
	return slice
}

func swap(slice []int, i int, j int) {
	temp := slice[i]
	slice[i] = slice[j]
	slice[j] = temp
}
